src/js/script.js

```diff
@@ -194,25 +224,82 @@
           }
         }
       }
+      thisProduct.priceSingle = price;
 
       price *=thisProduct.amountWidget.value;
 
       // update calculated price in the HTML
+
       thisProduct.priceElem.innerHTML = price;
     }
+    
+    addToCart(){
+      const thisProduct = this;
+
+      app.cart.add(thisProduct.prepareCartProduct());
+      
+    }
+
+    prepareCartProduct(){
+      const thisProduct = this;
+
+      const productSummary = {
+
+        id: thisProduct.id,
+        name: thisProduct.data.name,
+        amount: thisProduct.amountWidget.value,
+        priceSingle: thisProduct.priceSingle,
+        price: thisProduct.priceSingle * thisProduct.amountWidget.value,
+        params: thisProduct.prepareCartProductParams(),
+      };
+      
+      return(productSummary);
+    }
+
+    prepareCartProductParams() {
+      const thisProduct = this;
+    
+      const formData = utils.serializeFormToObject(thisProduct.form);
+      const params = {};
+    
+      // for very category (param)
+      for(let paramId in thisProduct.data.params) {
+        const param = thisProduct.data.params[paramId];
+    
+        // create category param in params const eg. params = { ingredients: { name: 'Ingredients', options: {}}}
+        params[paramId] = {
+          label: param.label,
+          options: {}
+        };
+    
+        // for every option in this category
+        for(let optionId in param.options) {
+          const option = param.options[optionId];
+          const optionSelected = formData[paramId] && formData[paramId].includes(optionId);
```

Comment:
Konwencja dla ułatwienia czytelności jest że jeżeli masz zmienne boolean to nazywają się zaczynając od is czyli isOptionSelected. Samo optionSelected daje wyraz że ten const przechowuje opcje która jest wybrana, a nie czy jest wybrana.